server/mailet/mailets/src/main/java/org/apache/james/transport/mailets/BayesianAnalysis.java

```diff
@@ -374,10 +374,10 @@ private String getAddressesString(Collection<MailAddress> addresses) {
         StringBuilder sb = new StringBuilder();
         sb.append('[');
         for (int i = 0; iter.hasNext(); i++) {
-            sb.append(iter.next());
-            if (i + 1 < addresses.size()) {
```

Comment:
```
i+1 < addresses.size()
i < addesses.size() - 1 
```

That seems very different from i > 0

IMO if you want to fix it, simply extract the size call in a variable out of the loop

Reply:
It looks like the purpose of the code is to insert ", " between each `MailAddress` returned by the iterator. In the existing code, ", " is appended after a `MailAddress`, except for after the last `MailAddress`. In the modified code, ", " is appended before the next `MailAddress`, if not the first `MailAddress`. Unless I am mistaken, the existing and modified codes have the same behavior.

Reply:
Right.

I would prefer a more explicit:

```
sb.append("[")
    .append(Joiner.on(',')
       .join(addresses))
    .append("]");
```

(Guava is available, so let's use it ;-) )

Reply:
That's a lot cleaner.

I have pushed a commit to use `Joiner`.